src/heif.h

```diff
@@ -273,6 +273,36 @@ void heif_context_free(struct heif_context*);
 
 struct heif_reading_options;
 
+enum heif_reader_offset {
+  heif_seek_start,
+  heif_seek_current,
+  heif_seek_end
+};
+
+struct heif_reader {
+  // API version supported by this reader
+  int reader_api_version;
+
+  // --- version 1 functions ---
+  uint64_t (*get_length)(struct heif_context* ctx, void* userdata);
```

Comment:
There might be inputs with unknown length or even inputs that grow while reading (e.g. displaying of partial images while downloading).
Do we really need the length explicitly? I think the current code does not use this.

Reply:
The `BitStreamRange` uses it to check if data can be read.

Reply:
Currently, it is checking the EOF flags.

Reply:
It also checks the remaining bytes:
https://github.com/strukturag/libheif/blob/a53a44e779e57b578324d465cf51dc4b40b0275a/src/bitstream.h#L66